gef.py

```diff
@@ -95,17 +96,22 @@
     # Compat Py2/3 hacks
     def range(*args):
         """Replace range() builtin with an iterator version."""
-        if len(args) < 1:
-            raise TypeError()
-        start, end, step = 0, args[0], 1
-        if len(args) == 2: start, end = args
-        if len(args) == 3: start, end, step = args
+        if len(args) == 0:
+            raise TypeError('range expected 1 arguments, got 0')
+        if len(args) >= 4:
+            raise TypeError('range expected at most 3 arguments, got 4')
+        start, stop, step = 0, args[0], 1
+        if len(args) == 2: start, stop = args
+        if len(args) == 3: start, stop, step = args
+        if step == 0:
+            raise ValueError('range arg 3 must not be zero')
+        pred = operator.ge if step > 0 else operator.le
```

Comment:
performance gain, maybe.

Reply:
Have you actually tested it? If so please provide some proofs.
By the look of it, it doesn't look like it, but I can be wrong.